rundeckapp/grails-app/services/rundeck/services/ReportService.groovy

```diff
@@ -38,55 +40,31 @@ class ReportService  {
     def grailsApplication
     AppAuthContextEvaluator rundeckAuthContextEvaluator
     ConfigurationService configurationService
+    ExecReportDataProvider execReportDataProvider
 
     static final String GRANTED_VIEW_HISTORY_JOBS = "granted_view_history_jobs"
     static final String DENIED_VIEW_HISTORY_JOBS = "rejected_view_history_jobs"
 
-    public Map reportExecutionResult(Map fields) {
+    public Map reportExecutionResult(SaveReportRequestImpl saveReportRequest) {
         /**
          * allowed fields are specified
          */
-
-        if (fields['rundeckEpochDateStarted']) {
-            def long dstart = Long.parseLong(fields['rundeckEpochDateStarted'])
-            if (dstart > 0) {
-                fields['dateStarted'] = new Date(dstart)
-            }
-        } else if (fields['epochDateStarted']) {
-            def long dstart = Long.parseLong(fields['epochDateStarted'])
-            if (dstart > 0) {
-                fields['dateStarted'] = new Date(dstart)
-            }
-        }
-        if (fields['rundeckEpochDateEnded']) {
-            def long dstart = Long.parseLong(fields['rundeckEpochDateEnded'])
-            if (dstart > 0) {
-                fields['dateCompleted'] = new Date(dstart)
-            }
-        } else if (fields['epochDateEnded']) {
-            def long dstart = Long.parseLong(fields['epochDateEnded'])
-            if (dstart > 0) {
-                fields['dateCompleted'] = new Date(dstart)
-            }
-        }
-
-        if (!fields['dateStarted']) {
-            fields['dateStarted'] = fields['dateCompleted']
+        if (!saveReportRequest.dateStarted) {
+            saveReportRequest.dateStarted= saveReportRequest.dateCompleted
         }
-        if(!fields.message){
-            fields.message="[no message]"
+        if(!saveReportRequest.message){
+            saveReportRequest.message="[no message]"
         }
-        fields.actionType= fields.status
-        def rep = new ExecReport(fields)
+        SaveReportResponseImpl saveReportResponse = execReportDataProvider.saveReport(saveReportRequest)
 
         //TODO: authorize event creation?
 
-        if (rep && !rep.save(flush: true)) {
+        if (saveReportResponse.report && !saveReportResponse.isSaved) {
```

Comment:
probably the intent of the original if statement was simply to check if the result was saved, the `rep &&` was unnecessary since the `rep` was created in the previous statement.  However, now the if clause could return `success:true` if the saveReportResponse.report is null.  I think this should just change to:
```suggestion
        if (!saveReportResponse.isSaved) {
```